columnflow/production/muon.py

```diff
@@ -0,0 +1,96 @@
+# coding: utf-8
+
+"""
+Muon related event weights.
+"""
+
+from columnflow.production import Producer, producer
+from columnflow.util import maybe_import
+from columnflow.columnar_util import set_ak_column, layout_ak_array
+
+np = maybe_import("numpy")
+ak = maybe_import("awkward")
+
+
+@producer(
+    used_columns={
+        "nMuon", "Muon.pt", "Muon.eta",
+    },
+    produces={
+        "muon_weight", "muon_weight_up", "muon_weight_down",
+    },
+)
+def muon_weights(self: Producer, events: ak.Array, **kwargs) -> ak.Array:
+    """
+    Reads the muon scale factor from the external file given in the config,
+    using the keys from the corresponding auxiliary entry in the config.
+    As of 10.2022, external files originating from a specific commit of
+    https://gitlab.cern.ch/cms-nanoAOD/jsonpog-integration/-/tree/master/POG/MUO
+
+    Example of external file in config:
+
+    .. code-block:: python
+
+        "muon_sf": ("/afs/cern.ch/user/m/mrieger/public/mirrors/jsonpog-integration-d0a522ea/POG/MUO/2017_UL/muon_z.json.gz", "v1"),  # noqa
+
+    Example of the corresponding auxiliary entry to read the correct sets from the json file:
+
+    .. code-block:: python
+
+        cfg.x.muon_sf_names = ("NUM_TightRelIso_DEN_TightIDandIPCut", "2017_UL")
+    """
+    if self.dataset_inst.is_data:
+        return events
+
+    # get year string
+    year = self.config_inst.x.muon_sf_names[1]
```

Comment:
maybe change the name to something else (e.g. `trig_year`) to make clear that this is (potentially) different to the campaign year?

Reply:
👍 I changed it to `sf_year`.